src/bootstrap/config.rs

```diff
@@ -1562,7 +1562,16 @@ impl Config {
 
     pub(crate) fn initial_rustfmt(&self) -> Option<PathBuf> {
         match &mut *self.initial_rustfmt.borrow_mut() {
-            RustfmtState::SystemToolchain(p) | RustfmtState::Downloaded(p) => Some(p.clone()),
+            RustfmtState::SystemToolchain(_) => {
+                let path = self.maybe_download_rustfmt();
+                if let Some(p) = &path {
+                    RustfmtState::Downloaded(p.clone())
+                } else {
+                    RustfmtState::Unavailable
+                };
+                path
+            }
+            RustfmtState::Downloaded(p) => Some(p.clone()),
```

Comment:
This not the right fix. We allow explicitly configuring a rustfmt in https://github.com/rust-lang/rust/blob/956ad918b9f6b923b897ffc9589c7200ceef7a9a/config.toml.example#L222-L224 and in that case we should continue to respect that option instead of downloading rustfmt.

Try looking around https://github.com/rust-lang/rust/blob/956ad918b9f6b923b897ffc9589c7200ceef7a9a/src/bootstrap/config.rs#L1128-L1134 instead.